internal/database/permission_sync_jobs_test.go

```diff
@@ -129,3 +114,150 @@ func TestPermissionSyncJobs_CreateAndList(t *testing.T) {
+	assert.Len(t, allJobs, 2)
+	assert.Equal(t, allJobs[0].ID, allJobs[0].UserID)
+	assert.Equal(t, allJobs[1].ID, allJobs[1].UserID)
+
+	// 3) Another low priority job without process_after for user1 is dropped
+	err = store.CreateUserSyncJob(ctx, 1, user1LowPrioJob)
+	assert.NoError(t, err)
+	allJobs, err = store.List(ctx, ListPermissionSyncJobOpts{})
+	assert.NoError(t, err)
+	// check that we still have 2 jobs. Job ID should match user ID.
+	assert.Len(t, allJobs, 2)
+	assert.Equal(t, allJobs[0].ID, allJobs[0].UserID)
+	assert.Equal(t, allJobs[1].ID, allJobs[1].UserID)
+
+	// 4) Insert some low priority jobs with process_after for both users. All of them should be inserted.
+	fiveMinutesLater := clock.Now().Add(5 * time.Minute)
+	tenMinutesLater := clock.Now().Add(10 * time.Minute)
+	user1LowPrioDelayedJob := PermissionSyncJobOpts{NextSyncAt: fiveMinutesLater, Reason: ReasonManualUserSync, TriggeredByUserID: user1.ID}
+	user2LowPrioDelayedJob := PermissionSyncJobOpts{NextSyncAt: tenMinutesLater, Reason: ReasonManualUserSync, TriggeredByUserID: user1.ID}
+	err = store.CreateUserSyncJob(ctx, 1, user1LowPrioDelayedJob)
+	assert.NoError(t, err)
+	err = store.CreateUserSyncJob(ctx, 2, user2LowPrioDelayedJob)
+	assert.NoError(t, err)
+	allDelayedJobs, err := store.List(ctx, ListPermissionSyncJobOpts{NotNullProcessAfter: true})
+	assert.NoError(t, err)
+	// check that we have 2 delayed jobs in total
+	assert.Len(t, allDelayedJobs, 2)
+	// userID of the job should be (jobID - 2)
+	assert.Equal(t, allDelayedJobs[0].UserID, allDelayedJobs[0].ID-2)
+	assert.Equal(t, allDelayedJobs[1].UserID, allDelayedJobs[1].ID-2)
+
+	// 5) Insert *high* priority job without process_after for user1. Check that low priority job is canceled
+	user1HighPrioJob := PermissionSyncJobOpts{HighPriority: true, Reason: ReasonManualUserSync, TriggeredByUserID: user1.ID}
+	err = store.CreateUserSyncJob(ctx, 1, user1HighPrioJob)
+	assert.NoError(t, err)
+	allUser1Jobs, err := store.List(ctx, ListPermissionSyncJobOpts{UserID: 1})
+	assert.NoError(t, err)
+	// check that we have 3 jobs for userID=1 in total (low prio (canceled), delayed, high prio)
+	assert.Len(t, allUser1Jobs, 3)
+	// check that low prio job (ID=1) is canceled and others are not
+	for _, job := range allUser1Jobs {
+		if job.ID == 1 {
+			assert.True(t, job.Cancel)
+		} else {
+			assert.False(t, job.Cancel)
+		}
+	}
+
+	// 6) Insert another low and high priority jobs without process_after for user1.
+	// Check that all of them are dropped since we already have a high prio job.
+	err = store.CreateUserSyncJob(ctx, 1, user1LowPrioJob)
+	assert.NoError(t, err)
+	err = store.CreateUserSyncJob(ctx, 1, user1HighPrioJob)
+	assert.NoError(t, err)
+	allUser1Jobs, err = store.List(ctx, ListPermissionSyncJobOpts{UserID: 1})
+	assert.NoError(t, err)
+	// check that we still have 3 jobs for userID=1 in total (low prio (canceled), delayed, high prio)
+	assert.Len(t, allUser1Jobs, 3)
```

Comment:
Nitpicky, I know, but: I think newlines make blocks like this more readable.

```suggestion
	err = store.CreateUserSyncJob(ctx, 1, user1LowPrioJob)
	assert.NoError(t, err)
	
	err = store.CreateUserSyncJob(ctx, 1, user1HighPrioJob)
	assert.NoError(t, err)
	
	allUser1Jobs, err = store.List(ctx, ListPermissionSyncJobOpts{UserID: 1})
	assert.NoError(t, err)
	// check that we still have 3 jobs for userID=1 in total (low prio (canceled), delayed, high prio)
	assert.Len(t, allUser1Jobs, 3)
```